.github/workflows/ci.yml

```diff
@@ -499,6 +499,70 @@ jobs:
           secret_vault_address: ${{ secrets.VAULT_ADDR }}
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
 
+  optimize-upgrade-tests:
+    if: needs.detect-changes.outputs.optimize-backend-changes == 'true'
+    needs: [detect-changes]
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+      - name: Setup Maven
+        uses: ./.github/actions/setup-maven
+        with:
+          secrets: ${{ toJSON(secrets) }}
+      - name: Login to Harbor registry
+        uses: ./.github/actions/login-registry
+        with:
+          secrets: ${{ toJSON(secrets) }}
```

Comment:
I'd like to assess if this step is needed: Which of the subsequent steps uses Docker images that need to be downloaded from registry.camunda.cloud?

In https://github.com/camunda/camunda/blob/main/.github/actions/compose/docker-compose.elasticsearch.yml the ElasticSearch image is downloaded from the official ES registry, not from registry.camunda.cloud.

I'm not sure what the other steps in this job do but if none of them needs registry.camunda.cloud you should remove this login step.

Reply:
Thanks, looks like I copied this over without needing to - removed now. Will do a bit more testing to see if the IT are too flaky for unified CI.